code/modules/atmospherics/machinery/components/unary_devices/thermomachine.dm

```diff
@@ -182,12 +197,16 @@
 	if(!port.total_moles()) // Nothing to cool? go home lad
 		return
 
+	// We perfectly can do W1+W2 / C1+C2 here but this lets us count the power easily.
 	var/port_capacity = port.heat_capacity()
 
 	// The difference between target and what we need to heat/cool. Positive if heating, negative if cooling.
 	var/temperature_target_delta = target_temperature - port.temperature
 
-	// We perfectly can do W1+W2 / C1+C2 here but this lets us count the power easily.
+	// Checking if the thermomachine is anomalous
+	if(istype(src, /obj/machinery/atmospherics/components/unary/thermomachine/anomalous) && temperature_target_delta < 0)
+		return
```

Comment:
Why are we doing this check? We should never need to check src's type, we know what it is because we're on it.

Reply:
Because without allows_cooling the only way to check the type to my knowledge is this way. I commented the reason earlier and left the conversation open.

Reply:
If there is a alternative, please let me know.

Reply:
Either have a var to block it or just let it work, I suppose.

Reply:
What do you mean limiting? If you have an idea of how strong it should be increasing the temp of existing thermomachines or otherwise let me know. I said it earlier but the numbers were picked very arbitrarily, if you have a better number I have no complaints changing it.

Reply:
> What do you mean limiting?

Limiting pyro anomalies to its own machine that can only heat, it should still just be a direct upgrade given to thermomachines because again, you're upgrading it with an anomaly core. You're not increasing the max cooling or anything. It's just a thermomachine that can now overheat.

Reply:
It is not limited to it's own machine, it has 2 new uses as mentioned earlier and in the updated description.

Reply:
They are very similar machines and I do not agree with the justification why they shouldn't be just be a heat upgrade to the base thermomachine on inserting the core.

I still will not be merging this myself while they're 2 separate machines.

Reply:
Understood and I disagree. There is no logical gymnastics I can do in my head to allow simply slotting something into a machine and allow it to be able to handle more than double it's normal max temperature. A dedicated machine would be far more efficient and thus I have made it so it is.

Reply:
> Understood and I disagree. There is no logical gymnastics I can do in my head to allow simply slotting something into a machine and allow it to be able to handle more than double it's normal max temperature.

"You hook up the pyro anomaly core to the src`s interface using some redundant cable connections to the heating element and it begins warming up"

Reply:
I like that idea, it makes far more sense and it could be a possible hint to the anomalous thermomachine by mentioning that the heating is inefficient.